ctru-rs/src/gfx.rs

```diff
@@ -138,6 +136,34 @@ impl TopScreen {
     pub fn get_wide_mode(&self) -> bool {
         unsafe { ctru_sys::gfxIsWide() }
     }
+
+    /// Returns a [`RawFrameBuffer`] for the given [`Side`] of the top screen.
+    ///
+    /// Note that the pointer of the framebuffer returned by this function can
+    /// change after each call to this function if double buffering is enabled.
+    pub fn get_raw_framebuffer(&mut self, side: Side) -> RawFrameBuffer {
+        RawFrameBuffer::for_screen_side(self.as_raw(), side.into())
+    }
+}
+
+impl BottomScreen {
+    /// Returns a [`RawFrameBuffer`] for the bottom screen.
+    ///
+    /// Note that the pointer of the framebuffer returned by this function can
+    /// change after each call to this function if double buffering is enabled.
+    pub fn get_raw_framebuffer(&mut self) -> RawFrameBuffer {
+        RawFrameBuffer::for_screen_side(self.as_raw(), Side::Left.into())
+    }
+}
+
+impl RawFrameBuffer {
+    fn for_screen_side(screen: ctru_sys::gfxScreen_t, side: ctru_sys::gfx3dSide_t) -> Self {
+        let mut buf = RawFrameBuffer::default();
```

Comment:
Instead of creating a default and then modifying it, you can just construct the struct at the end of the function. This also avoids giving the struct a `Default` impl, which is callable by users (kind of misleading).

Reply:
Yes, remove the `Default` and just use the struct literal instead.